### DISCUSSION TITLE
[ZK Stack] - How to acheive interchain operatability?

### DISCUSSION THREAD

**Discussion by sachanayush47** [zkSync-Community-Hub/zksync-developers#780] [ZK Stack] [answered]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [ ] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.82.0

### Operating System

- [X] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

We have multiple chains running under ecosystem with different configuration. We want to achieve interoperability for enhanced UX, users should be able to deposit and withdraw asset. We came to know about 2 things: [HyperBridge](https://docs.zksync.io/zk-stack/concepts/zk-chains#how-hyperbridges-work) and [Interop](https://github.com/matter-labs/zksync-era/blob/main/docs/src/specs/zk_chains/interop.md) but we are not sure which one is exactly used HyperBridge or Interop as both seems to offer similar functionality and what all things we need to setup to have this?

### Expected Behavior

NA

### Current Behavior

NA

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by alhenaai** [comment]
To achieve interchain operability within your ecosystem using ZKsync, you can explore both the HyperBridge and Interop functionalities, as they cater to different aspects of cross-chain interactions:

1. **HyperBridge**: This is designed for verifying transactions across chains using Merkle proofs. It involves a series of steps including transaction initiation on a ZK chain, settlement on Ethereum's Layer 1, and transaction verification and execution on the receiving ZK chain. This process is ideal for unifying liquidity and ensuring secure asset transfers across different chains within the ecosystem.

2. **Interop**: This functionality allows for seamless interaction between different chains, although it does not synchronize the chains fully (e.g., the same address may have different balances on different chains). Interop is more about enabling communication and interaction rather than mirroring state or balances across chains.

For setting up these functionalities, you will need to:
- Ensure that your system contracts and smart contracts are capable of handling cross-chain transactions.
- Set up the necessary infrastructure on your server and possibly adjust your zkEVM configurations to support these features.

For detailed guidance on setting up and choosing between HyperBridge and Interop based on your specific needs, you might want to refer to the respective documentation on [HyperBridge](https://docs.zksync.io/zk-stack/concepts/zk-chains#how-hyperbridges-work) and [Interop](https://github.com/matter-labs/zksync-era/blob/main/docs/src/specs/zk_chains/interop.md).

For more in-depth discussion and community support, you can visit the [ZKsync developers discussions on Github](https://github.com/zkSync-Community-Hub/zksync-developers/discussions).

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by MexicanAce** [comment, marked as answer]
Hi @sachanayush47 👋 , Interop functionality is under active development and there's currently a working alpha version for ZKsync Era Sepolia Testnet and Sophon (a ZK chain) Testnet.

However, for local ecosystems and local development, the team is working on updates to `zkstack` and local dev for others to also try out Interop. We'll post updates in our docs and on our [socials](https://x.com/ZKsync) when this functionality is more readily available.

**Comment by Satyam78612** [comment]
His @sachanayush47, I am building a L2 appchain using zksync stack and we want to create a wallet on our appchain along with the particle network. I would like to ask that particle network will be compatible with our appchain or not?

**Reply by sachanayush47** [reply]
Hi, I have never tried particle network so Im not sure about this.

**Reply by Satyam78612** [reply]
@sachanayush47 Is here anyone who can help me?

**Reply by sachanayush47** [reply]
You can create a new discussion thread.
